rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -799,8 +799,9 @@ object Reduce {
         } else {
           for {
             exprEvaled <- evalExpr(p)
-            _          <- costAccountingAlg.charge(toByteArrayCost(exprEvaled))
-            ba         <- s.fromEither(serialize(exprEvaled))
+            exprSubst  <- substituteAndCharge[Par, M](exprEvaled, 0, env, costAccountingAlg)
```

Comment:
1. Don't we want all method arguments (including "this") be fully substituted before call/execution? Or do we want lazy semantics wherever possible?

2. Do we have in plans some form of method call protocol, and API for it? I'd imagine defining a method should be as easy as *declaring* the signature (including parameter strictness) and body. I'm not saying we should do this in this PR, just asking a question.

Reply:
1. No. Most methods should be fine without fully substituted terms.
2. I don't have any plans to do so.